lib/elixir/lib/string.ex

```diff
@@ -1504,6 +1504,20 @@ defmodule String do
       iex> String.next_codepoint("olá")
       {"o", "lá"}
 
+  Note, however, that getting the next codepoint via pattern
+  matching is generally preferred. For example
```

Comment:
I like this but I am not sure if we should say "preferred". Maybe we should just say it is possible? What does everyone think?

Reply:
I wanted to say something like "preferred if possible", "preferred if available"... something like that, but didn't find the right words (for my taste) and left it short.

Reply:
Although... it is always possible now that I think, since that is what the function does really.

The use case I had in mind was the one in the examples mainly.

Reply:
"can be simplified"

Reply:
"can be simplified"

Reply:
@fxn to be honest, I'm not entirely sure. I suspect the goal was to offer similar & complete API for both codepoints & graphemes.

Reply:
Cool!

What about a remark that says something in the line of

> This function is provided for symmetry with the API for graphemes, but doing the same with pattern matching is simpler and more performant. For example... < function definitions in both styles >

?

Reply:
I am not sure if that is relevant for users. Let's go with the simplest change here and just mention it can be implemented more efficiently with pattern matching too. :)

Reply:
In my view it is relevant because if you put yourself in the shoes of a user reading:

> Function _F_ does this, but the same can be done more efficiently this way.

and the presented alternative it is not only more performant, but also looks simpler, and idiomatic, the user will inevitably wonder what is the point of the function, and more importantly _when_ should they use this apparently "inferior" option.

Reply:
Alright, I am convinced. :) Let's ship what you proposed!

Reply:
However, both examples are not "exactly" the same: in the pattern matching version the value assigned to the `cp` variable is an integer codepoint, and in the `String.next_codepoint/1` version is a string codepoint. 

Maybe we should mention that? or update the pattern matching example to turn the integer codepoint into a string one?

Reply:
@fertapric I'll take that into account. Let me come with a reworded patch.